tests/headers/packed-align-conflict.h

```diff
@@ -1,3 +1,5 @@
-struct FndrOpaqueInfo {
-  char opaque[16];
+typedef unsigned short uint16_t;
```

Comment:
I don't think this is right. This works around the issue by making alignment bigger naturally, but for `char` it still behaves incorrectly. I think we should fix the underaligned member behavior.

Reply:
Ah, I got what you meant.

I thought the issue is related to 2 unrelated bugs. One for codegen and the other one for align.
But I now understand `#[align(2)]` itself generate correctly aligned and padded.

For me fixing only codegen is also worth to do because the other part can be manually fixed by opaque_type, but it could be worse by point of view by having silently buggy layout.

I'd like to suggest a new desired bindgen result:
```rust
pub struct AlignedAndPacked(__AlignedAndPacked_Aligned);

#[repr(C)]
#[repr(align(2))]
#[derive(Debug, Default, Copy, Clone)]
pub struct __AlignedAndPacked_Aligned {
    pub opaque: [u16; 8usize],
}
```

And its working test:
```rust
#[test]
fn bindgen_test_layout_AlignedAndPacked() {
    const UNINIT: ::std::mem::MaybeUninit<AlignedAndPacked> =
        ::std::mem::MaybeUninit::uninit();
    let ptr = UNINIT.as_ptr();
    assert_eq!(
        ::std::mem::size_of::<AlignedAndPacked>(),
        16usize,
        concat!("Size of: ", stringify!(AlignedAndPacked))
    );
    assert_eq!(
        ::std::mem::align_of::<AlignedAndPacked>(),
        2usize,
        concat!("Alignment of ", stringify!(AlignedAndPacked))
    );
    assert_eq!(
        unsafe { ::std::ptr::addr_of!((*ptr).0) as usize - ptr as usize },
        0usize,
        concat!(
            "Offset of field: ",
            stringify!(AlignedAndPacked),
            "::",
            stringify!(opaque)
        )
    );
}
```

Reply:
I think I am missing a simple thing. If `repr(align(2))` always add padding to make it aligned, there is no chance it cannot be packed. Lets add aligned but remove packed.

Reply:
Not really tho, right? I mean, for this case, sure, but for more complex cases, you might have padding between members...

Reply:
No, this one was wrong assumption. I made more changes below and I think the new one is correct including padding between members. The new approach covers both padding between members and packed members. The last piece I have to know is explicit align flag of the type.